templates/api.yaml

```diff
@@ -79,17 +88,893 @@ Resources:
+                EventName:
+                  type: "string"
+                  description: "The Job Event name"
+                EventData:
+                  type: "object"
+                  description: "Free form field containing data about the event. Structure varies based on the event"
+                EmitterId:
+                  type: "string"
+                  description: "The identifier for the service or service instance which emitted the event"
+                  example: "StepFunctions"
+                Expires:
+                  type: "integer"
+                  description: "Expiry date when the item will be deleted as Epoch time"
+                Sk:
+                  type: "string"
+                  description: "Internal field used as part of DynamoDB single table design"
+                Type:
+                  type: "string"
+                  enum:
+                    - "Job"
+                  description: "Internal field used as part of DynamoDB single table design"
+            Settings:
+              description: "A Settings object"
+              type: "object"
+              properties:
+                AthenaConcurrencyLimit:
+                  type: "integer"
+                  description: "Athena concurrency setting for this job"
+                AthenaQueryMaxRetries:
+                  type: "integer"
+                  description: "Max number of retries to each Athena query after a failure"
+                DeletionTasksMaxNumber:
+                  type: "integer"
+                  description: "Max Fargate tasks setting for this job"
+                ForgetQueueWaitSeconds:
+                  type: "integer"
+                  description: "Forget queue wait setting for this job"
+                QueryExecutionWaitSeconds:
+                  type: "integer"
+                  description: "Query execution wait setting for this job"
+                QueryQueueWaitSeconds:
+                  type: "integer"
+                  description: "Query queue worker wait setting for this job"
+              required:
+                - "AthenaConcurrencyLimit"
+                - "AthenaQueryMaxRetries"
+                - "DeletionTasksMaxNumber"
+                - "ForgetQueueWaitSeconds"
+                - "QueryExecutionWaitSeconds"
+                - "QueryQueueWaitSeconds"
+            Error:
+              description: "A standard error object"
+              type: "object"
+              properties:
+                Message:
+                  description: "Error message"
+                  type: "string"
+              required:
+                - "Message"
+          responses:
+            InvalidRequest:
+              description: "Error response for invalid requests"
+              content:
+                application/json:
+                  schema:
+                    $ref: '#/components/schemas/Error'
+          securitySchemes:
+            CognitoAuthorizer:
```

Comment:
probably should change this name as well because cognito might not be in use

Reply:
Agreed